### DISCUSSION TITLE
Notify user on unsuccesful pulls when offline

### DISCUSSION THREAD

**Discussion by zaeph** [magit/forge#671] [Ideas]
Hello,

Now that I've got `forge` working on my system, I've been able to try things out.  One of the questions I had was whether `forge` would allow me to work on issues whilst offline.

I've tried pushing an issue to a tracker when offline.  The only indication that the push hasn't gone through is that the buffer isn't closed: no informative message.  It might be useful to notify the user on those fails.

(**Side-note**: If we close the buffer after saving it, running `forge-create-issue` again asks you whether you want to edit the previous draft or thrash it.  This is great, but maybe we could expand this behaviour to have multiple concurrent drafts?)

Then, I tried `forge-pull` when offline.  It displays `Pulling $user/$repo` in the modeline, but it doesn't seem to time-out, or if it does, I haven't witnessed it (waited ~2 min).

`magit-fetch-from-pushremote` when offline lets you know that the pull didn't happen.  In `magit-process-buffer`:
```
128 git … fetch origin

fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

Not entirely informative, but I guess it'd be hard to ascertain that there's a connection problem.

**Comment by tarsius** [comment]
I don't plan to do this, but because it is not completely off the table, I have converted it to a discussion, instead of only closing the issue.

**Comment by tarsius** [comment]
> it doesn't seem to time-out, or if it does, I haven't witnessed it (waited ~2 min).

`url-retrieve` eventually times out. There's probably an option to instruct it to time out faster.

----

To be able to focus on other projects, I have to reduce the time I spend on Forge. That involves deciding to not implement certain features and I'm afraid this is one of those. Sorry.
